src/app/ui/earn/EarnOpportunitiesAll/EarnOpportunitiesAllServer.tsx

```diff
@@ -0,0 +1,18 @@
+import { FC, Suspense } from 'react';
+import { EarnOpportunitiesAllClient } from './EarnOpportunitiesAllClient';
+import { EarnOpportunityFilter } from 'src/app/lib/getOpportunitiesFiltered';
+import { EarnOpportunitiesAllSkeleton } from './EarnOpportunitiesAllSkeleton';
+
+interface EarnOpportunitiesAllServerProps {
+  initialFilters: EarnOpportunityFilter;
+}
+
+export const EarnOpportunitiesAllServer: FC<
+  EarnOpportunitiesAllServerProps
+> = async ({ initialFilters }) => {
+  return (
+    <Suspense fallback={<EarnOpportunitiesAllSkeleton />}>
+      <EarnOpportunitiesAllClient initialFilters={initialFilters} />
+    </Suspense>
+  );
+};
```

Comment:
From what I understand we're adding 3 files to render skeletons on the server side (we don't get data on the server yet, so we're just moving skeletons).

Could we use something like https://nextjs.org/docs/app/api-reference/file-conventions/loading instead? That seems more nextjs'y

Reply:
Totally agree, though loading.ts isn't currently used in our app, so I'd prefer sticking with our existing pattern for consistency. We could explore this in a separate PR.
And I think modular skeleton components for specific page sections makes sense, as this makes it easier to update components independently or migrate parts of the page to SSR.